test/test_distributions.py

```diff
@@ -962,3 +962,16 @@ def test_generated_sample_distribution(jax_dist, sp_dist, params,
         our_samples = jax_dist.sample(key, (N_sample,))
         ks_result = osp.kstest(our_samples, sp_dist(*params).cdf)
         assert ks_result.pvalue > 0.05
+
+
+@pytest.mark.parametrize('jax_dist, params, support', [
+    (dist.BernoulliLogits, (5.,), np.arange(2)),
+    (dist.BernoulliProbs, (0.5,), np.arange(2)),
+    (dist.BinomialLogits, (4.5, 10), np.arange(11)),
+    (dist.BinomialProbs, (0.5, 11), np.arange(12)),
+    (dist.BetaBinomial, (2., 0.5, 12), np.arange(13)),
+    (dist.CategoricalLogits, (np.array([3., 4., 5.]),), np.arange(3)),
+    (dist.CategoricalProbs, (np.array([[0.1, 0.5, 0.4]]),), np.arange(3)),
```

Comment:
The shape here should be `(3, 1)` right for the additional batch dim? Maybe we can add some shape asserts too in the test.

Reply:
I am not sure if we need to add additional dimensions for `enumerate_support`. Based on [Pyro implementation](https://github.com/pyro-ppl/pyro/pull/2307/files#diff-70ab0ab836b4c309f534671a06651eeaR97), it seems to be not required.

Reply:
Hmm..the distribution's `enumerate_support` takes into account batch dims.
e.g.
```python
In [4]: dist.Categorical(torch.tensor([[0.1, 0.9]])).enumerate_support()
Out[4]:
tensor([[0],
        [1]])

In [5]: dist.Categorical(torch.tensor([0.1, 0.9])).enumerate_support()
Out[5]: tensor([0, 1])
```

Reply:
Yes, but with `expand=False`, it just appends [singleton batch dimensions](https://github.com/pytorch/pytorch/blob/master/torch/distributions/categorical.py#L124). With @eb8680 's implementation, those singleton dimensions are squeezed.

Reply:
I see. I think there is benefit in keeping the distribution's interface separate and consistent with pytorch's shape semantics. We can munge these shapes for the purpose of inference if needed. An alternative is to not expose this method in the distribution's interface and just provide a private utility function to aid @eb8680's refactor (making use of the integer interval support for the distribution's instance).

Reply:
I agree with @neerajprad, best to stick closely to PyTorch distribution semantics for `enumerate_support`.

Reply:
IMO if inhomogeneous enumerate support is not supported, there is no reason to match batch_shape. It is not important so I'll add `expand` argument for consistency.

Reply:
> IMO if inhomogeneous enumerate support is not supported, there is no reason to match batch_shape.

The bigger-picture rationale for matching `batch_shape` in `enumerate_support` is that other algorithms (e.g. Pyro's `TraceTMC_ELBO`) produce sample tensors that are not homogeneous, and we want to be able to reuse the same filtering and variable elimination code across as many such algorithms as possible.

Reply:
Thanks for your inputs, @neerajprad @eb8680 !